core-bundle/src/Resources/contao/controllers/BackendMain.php

```diff
@@ -135,7 +135,7 @@ public function run()
 			$session['backend_modules'][Input::get('mtg')] = (isset($session['backend_modules'][Input::get('mtg')]) && $session['backend_modules'][Input::get('mtg')] == 0) ? 1 : 0;
 			$objSessionBag->replace($session);
 
-			Controller::redirect(preg_replace('/(&(amp;)?|\?)mtg=[^& ]*/i', '', Environment::get('request')));
+			Controller::redirect(preg_replace('/(&(amp;)?|\?)mtg=[^& ]*$|mtg=[^&]*&(amp;)?/i', '', Environment::get('request')));
```

Comment:
```suggestion
			Controller::redirect(preg_replace('/(&(amp;)?|\?)mtg=[^& ]*($|&(amp;)?)/i', '', Environment::get('request')));
```

Reply:
that will not work. That would remove the leading and ending ampersand if both are present